### PR TITLE
Standardize and test support for multiple docker images

### PR DESCRIPTION
**Pull request by jakirkham**
Prefer the first entry in `docker_image` as opposed to the last one. This is nearly always arbitrary as `docker_image` should contain one value. However this fixes an issue when the CUDA images are included by ensuring only the `condaforge/linux-anvil-comp7` image gets picked.

cc @isuruf

xref: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/285

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by isuruf** [comment]
This still adds the other `docker_image` names into feedstocks without using cuda which is unnecessary IMO. Could you make sure the `docker_image` is a list of one in maybe `finalize_config`? You can remove this `0` and `-1` change as well.

**Comment by jakirkham** [comment]
Sounds reasonable. Have pushed a new commit, which does this. Seems to have the right behavior for both a normal feedstock and one using CUDA.

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -365,8 +365,11 @@ def finalize_config(config, platform, forge_config):
     """For configs without essential parameters like docker_image
     add fallback value.
     """
-    if platform.startswith("linux") and not "docker_image" in config:
-        config["docker_image"] = [forge_config["docker"]["fallback_image"]]
+    if platform.startswith("linux"):
+        if "docker_image" in config:
+            config["docker_image"] = config["docker_image"][0]
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py]: Can you make this a list? There's no reason to change this. (I think conda-build will complain as well)

Comment (jakirkham) [reply conda_smithy/configure_feedstock.py]: I'm confused. Won't we need to index the value in the CI scripts then?

Comment (isuruf) [reply conda_smithy/configure_feedstock.py]: What I would like to see is that https://github.com/conda-forge/zlib-feedstock/pull/35 will be a no-op. Can you rerender that PR with this branch again?

Comment (jakirkham) [reply conda_smithy/configure_feedstock.py]: Ok, thanks for clarifying. Will do. Have been doing this locally as we have been iterating.

Comment (jakirkham) [reply conda_smithy/configure_feedstock.py]: The latest commit makes re-rendering `zlib` a no-op.

**Comment by isuruf** [comment]
Looks good to me. Can you add a test?

**Comment by jakirkham** [comment]
Sure. Have pushed a test and a news entry. Please let me know if there is anything else. 🙂

**Comment by jakirkham** [comment]
Also thanks for the reviews here.

**Comment by isuruf** [comment]
Thanks @jakirkham. Can you add a test that when cuda is not used, the first value is used?

File: conda_smithy/templates/azure-pipelines-linux.yml.tmpl [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
       {{ config_name }}:
         CONFIG: {{ config_name }}
         UPLOAD_PACKAGES: {{ upload }}
-        DOCKER_IMAGE: {{ config["docker_image"][-1] }}
+        DOCKER_IMAGE: {{ config["docker_image"][0] }}
```
Comment (isuruf) [inline conda_smithy/templates/azure-pipelines-linux.yml.tmpl]: Can you revert this change to reduce the diff?

Comment (jakirkham) [reply conda_smithy/templates/azure-pipelines-linux.yml.tmpl]: Sure, done.

**Comment by jakirkham** [comment]
> Can you add a test that when cuda is not used, the first value is used?

So this is testing both. CUDA is used for 3 cases and is not used for 1.

**Comment by isuruf** [comment]
They are two different cases. When you use cuda, `docker_image` list of 4 is split into a list of 1 each. You are testing it here. When you don't use cuda, `docker_image` is a list of 4 and is not split, but is changed to a list of 1 in `finalize_config`. This is not tested.

**Comment by jakirkham** [comment]
Alright how about this test?

**Comment by jakirkham** [comment]
Thanks @isuruf 🙂
